pandas/tests/reshape/test_util.py

```diff
@@ -71,7 +71,13 @@ def test_invalid_input(self, X):
 
     def test_exceed_product_space(self):
         # GH31355: raise useful error when produce space is too large
-        msg = "Product space too large to allocate arrays!"
+        msg = (
```

Comment:
I suppose it's okay to match against the full error message here, although the test becomes a tad more brittle